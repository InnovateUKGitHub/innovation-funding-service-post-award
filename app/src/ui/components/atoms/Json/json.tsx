interface JsonProps {
  value: unknown;
}

export const Json: React.FunctionComponent<JsonProps> = ({ value }: JsonProps) => {
  return <pre>{JSON.stringify(value, null, 5)}</pre>;
};
