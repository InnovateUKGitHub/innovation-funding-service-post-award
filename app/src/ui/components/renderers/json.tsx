import React from "react";

export const Json: React.SFC<{value: any}> = (props) => {
    return <pre>{JSON.stringify(props.value, null, 5)}</pre>;
};
