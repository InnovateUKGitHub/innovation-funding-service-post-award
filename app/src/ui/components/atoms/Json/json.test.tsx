import { render } from "@testing-library/react";
import { Json } from "./json";

describe("Json", () => {
  it("should render a JSON stringified form of the object passed in", () => {
    expect(render(<Json value={{ foo: "bar", baz: 69, quux: true }} />).container).toMatchSnapshot();
  });
});
