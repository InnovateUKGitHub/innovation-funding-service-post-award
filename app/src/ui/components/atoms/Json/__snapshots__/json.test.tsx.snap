// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Json should render a JSON stringified form of the object passed in 1`] = `
<div>
  <pre>
    {
     "foo": "bar",
     "baz": 69,
     "quux": true
}
  </pre>
</div>
`;
